tests/phpunit/TestFramework/PhpUnit/Coverage/IndexXmlCoverageParserTest.php

```diff
@@ -241,49 +274,31 @@ public function noCoveredLineReportProviders(): Generator
         ];
     }
 
-    public function test_it_errors_when_the_source_file_could_not_be_found(): void
-    {
-        $incorrectCoverageSrcDir = Path::canonicalize(XmlCoverageFixtures::FIXTURES_INCORRECT_COVERAGE_DIR . '/src');
+    /**
+     * @param iterable<string, XmlCoverageFixture> $coverageFixtures
+     * @param iterable<string, SourceFileData> $sourceFilesData
+     */
+    private function assertCoverageFixtureSame(
+        iterable $coverageFixtures,
+        iterable $sourceFilesData
+    ): void {
+        $normalizedFixtures = [];
 
-        // Replaces dummy source path with the real path
-        $xml = preg_replace(
-            '/(source=\").*?(\")/',
-            sprintf('$1%s$2', $incorrectCoverageSrcDir),
-            file_get_contents(XmlCoverageFixtures::FIXTURES_INCORRECT_COVERAGE_DIR . '/coverage-xml/index.xml')
-        );
+        foreach ($coverageFixtures as $fixture) {
+            /* @var XmlCoverageFixture $fixture */
+            $normalizedFixtures[$fixture->sourceFilePath] = $fixture->serializedCoverage;
+        }
 
-        try {
-            $this->parser->parse('/path/to/index.xml', $xml);
+        $normalizedFilesData = [];
 
-            $this->fail();
-        } catch (InvalidCoverage $exception) {
-            $this->assertSame(
-                sprintf(
-                    'Could not find the source file "%s/zeroLevel.php" referred by '
-                    . '"%s/zeroLevel.php.xml". Make sure the coverage used is up to date',
-                    $incorrectCoverageSrcDir,
-                    Path::canonicalize(XmlCoverageFixtures::FIXTURES_COVERAGE_DIR)
-                ),
-                $exception->getMessage()
-            );
-            $this->assertSame(0, $exception->getCode());
-            $this->assertNull($exception->getPrevious());
+        foreach ($sourceFilesData as $data) {
+            /* @var SourceFileData $data */
+            $normalizedFilesData[$data->getSplFileInfo()->getRealPath()] = $data->retrieveCoverageReport();
```

Comment:
Not sure to follow you: that transformation here is to do the mapping which is not the job of `convertToArray()` IMO

Reply:
It did that before. The method's signature is unchanged, it still accepts two iterables. `convertToArray` will be happy to accept an iterable, as it was. So what's for these two arrays?

Reply:
I'm not doing the conversion for transforming it into an array, but for correcting the mapping because here what we compare is the coverage reports, not the source file info datas. This is an alternative to have an adapter and require to use the adapter everywhere in this test

Reply:
This method was doing the same job before, and didn't buffer one of the outputs. Now it buffers both. This is inefficient.

Reply:
@sanmai I think you misread the code there. I'm not doing that for the buffering, but for the mapping which was previously done in the "LegacyAdapter""